content/articles/2017-07-ggplot2-tidy-evaluation.Rmarkdown

```diff
@@ -0,0 +1,150 @@
+
+```{r aes-quo}
+x_var <- quo(cyl)
+y_var <- quo(mpg)
+
+by_cyl <- mtcars %>%
+  group_by(!!x_var) %>%
+  summarise(mean = mean(!!y_var))
+
+ggplot(by_cyl, aes(!!x_var, mean)) +
+  geom_point()
+```
+
+You can now use [quasiquotation](https://adv-r.hadley.nz/quasiquotation.html) in [`aes()`](https://ggplot2.tidyverse.org/reference/aes.html), [`facet_wrap()`](https://ggplot2.tidyverse.org/reference/facet_wrap.html), and [`facet_grid()`](https://ggplot2.tidyverse.org/reference/facet_grid.html). For `aes()`, quasiquotation (`!!`, `!!!`, `:=`) replaces [`aes_()`](https://ggplot2.tidyverse.org/reference/aes_.html) and [`aes_string()`](https://ggplot2.tidyverse.org/reference/aes_.html) (though these functions are being soft deprecated, and will be around for a while).
+
+Here we'll use quasiquotation in place of `aes_()` to make a function to make a pie chart that allows the user to avoid using quotation marks when they pass their arguments to the function.[^1] The initial function, below, would require that the user know the exact specifications to pass to `aes()`.
+
+```{r piechart-basic}
+piechart_basic <- function(data, mapping) {
+  ggplot(data, mapping) +
+    geom_bar(width = 1) +
+    coord_polar(theta = "y") +
+    xlab(NULL) +
+    ylab(NULL)
+}
+piechart_basic(mpg, aes(factor(1), fill = class))
+```
+
+By quoting and unquoting the arguments, the user can leave the arguments bare when calling the function.
+
+```{r piechart-qq}
+piechart <- function(data, var, ...) {
+  var <- enquo(var)
+  piechart_basic(data, aes(!!factor(1), fill = !!var))
+}
+piechart(mpg, class)
+```
+
+The basic pattern is to enquote and unquote named arguments.
+
+```{r scatter-by}
+scatter_by <- function(data, x, y) {
+  x <- enquo(x)
+  y <- enquo(y)
+
+  ggplot(data) + geom_point(aes(!!x, !!y))
+}
+scatter_by(mtcars, disp, drat)
+```
+
+### Facetting with `vars()`
+
+To support quasiquotation in facetting we’ve added a new helper function: [`vars()`](https://ggplot2.tidyverse.org/reference/vars.html), short for variables. Instead of `facet_grid(x + y ~ a + b)` you can now write `facet_grid(vars(x, y), vars(a, b))`. The formula interface won’t go away; but the new `vars()` interface supports tidy evaluation, so can be easily programmed with.
+
+`vars()` is used to supply variables or expressions, evaluated in the context of the dataset to form facetting groups.
+
+```{r facet-vars}
+p <- ggplot(mpg, aes(displ, cty)) + geom_point()
+
+p + facet_grid(rows = vars(drv))
+```
+
+Using quosures ensures that the variable comes from the context of the dataframe. Since the ellipsis is the first and only argument of `vars()`, you can use unquote splicing with the `!!!` operator to pass in a list of arguments.
```

Comment:
Replace para with something like:

Compared to `aes()`, `vars()` takes unnamed arguments. This makes it a more natural fit for use with `!!!`, the unquote-splice operator.